zproject/settings.py

```diff
@@ -320,6 +321,7 @@ def get_secret(key):
     'django.middleware.locale.LocaleMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'zerver.middleware.SetRemoteAddrFromForwardedFor',
```

Comment:
So for the same reason TagRequests is a dependency of LogRequest, SetRemote.... is as well. The request should be fully built before LogRequests does it's work on it. I understand what is it doing now.